src/test/java/io/confluent/idesidecar/restapi/models/graph/RealLocalFetcherTest.java

```diff
@@ -191,4 +195,42 @@ void assertExtractLocalhostAddresses(String input, String...expectedAddresses) {
     assertEquals(Arrays.asList(expectedAddresses), addresses);
   }
 
+  @Test
+  void schemaRegistryFetch_shouldParseConfigResponseFromValidJson() {
+    var json = loadResource(
+        "confluent-local-resources-mock-responses/get-schema-registry-response.json"
+    );
+    SchemaRegistryConfigResponse config = localFetcher.parseSchemaRegistryConfig(URL, json);
+    assertEquals("BACKWARD", config.compatibilityLevel());
+
+  }
+
+  @Test
+  void shouldReturnValidSchemaRegistryUri() throws CreateConnectionException {
+    var localConfig = new ConnectionSpec.LocalConfig("http://localhost:8085");
+    var connectionSpec = new ConnectionSpec("1", "Local Connection", ConnectionSpec.ConnectionType.LOCAL, null, localConfig);
+    manager.createConnectionState(connectionSpec);
+
+    String uri = localFetcher.resolveSchemaRegistryUri("1");
+    assertEquals("http://localhost:8085", uri);
+  }
+
+  @Test
+  void shouldReturnDefaultSchemaRegistryUriWhenLocalConfigIsNull() throws CreateConnectionException {
+    var connectionSpec = new ConnectionSpec("2", "Local Connection", ConnectionSpec.ConnectionType.LOCAL, null, null);
+    manager.createConnectionState(connectionSpec);
+
+    String uri = localFetcher.resolveSchemaRegistryUri("2");
+    assertEquals("http://localhost:8081", uri);
+  }
+
+  @Test
+  void shouldReturnNullWhenSchemaRegistryUriIsBlank() throws CreateConnectionException {
+    var localConfig = new ConnectionSpec.LocalConfig("");
+    var connectionSpec = new ConnectionSpec("3", "Local Connection", ConnectionSpec.ConnectionType.LOCAL, null, localConfig);
+    manager.createConnectionState(connectionSpec);
+
+    String uri = localFetcher.resolveSchemaRegistryUri("3");
+    assertNull(uri); // No Schema Registry
+  }
```

Comment:
We should also have a test for when the SR URI is null
```suggestion
  }

  @Test
  void shouldReturnDefaultSchemaRegistryUriWhenSchemaRegistryUriIsNull() throws CreateConnectionException {
    var localConfig = new ConnectionSpec.LocalConfig(null);
    var connectionSpec = new ConnectionSpec("4", "Local Connection", ConnectionSpec.ConnectionType.LOCAL, null, localConfig);
    manager.createConnectionState(connectionSpec);

    String uri = localFetcher.resolveSchemaRegistryUri("4");
    assertEquals("http://localhost:8081", uri);
  }
```

Reply:
Done.